### PR TITLE
[4.x] Replace `jenssegers/agent` and use latest `mobiledetect/mobiledetectlib`

### PR DESCRIPTION
**Pull request by crynobone**
* Port `Jenssegers\Agent\Agent` to `Laravel\Jetstream\Agent`
  - Without `isRobot()` detection

### CODE REVIEW

File: src/Agent.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+     *
+     * @return bool
+     */
+    public function isDesktop()
+    {
+        // Check specifically for cloudfront headers if the useragent === 'Amazon CloudFront'
+        if (
+            $this->getUserAgent() === static::$cloudFrontUA
+            && $this->getHttpHeader('HTTP_CLOUDFRONT_IS_DESKTOP_VIEWER') === 'true'
+        ) {
```
Comment (crynobone) [inline src/Agent.php:85]: This requires https://github.com/serbanghita/Mobile-Detect/blob/b73a5bdbcec74a6d33389cc8fe3a5af5d0965d01/src/MobileDetect.php#L1036-L1038 to actually work.

**Review by driesvints — APPROVED** [review]
Amazing work @crynobone! 👏

**Comment by Sarsmooch** [comment]
I don't know if I could be doing something wrong, but after the update, `$session->agent->platform()` and `$session->agent->browser()` return just 1. I created a new empty laravel project to make sure of this, and the same thing happens.

**Comment by Jared87** [comment]
> I don't know if I could be doing something wrong, but after the update, `$session->agent->platform()` and `$session->agent->browser()` return just 1. I created a new empty laravel project to make sure of this, and the same thing happens.
> 
> What seems even stranger to me is that during testing, when doing `{{ $session->agent->platform() }}`, this prints the platform correctly, while `{{ $session->agent->platform() ? $session->agent->platform() : __('Unknown') }}` prints '1' (true)

I have the same issue, as i see the problem seems to occur somewhere in the caching, but i cannot say it in detail. 
The Agent::retrieveUsingCacheOrResolve() Method has the correct result, but the return value seems to be true.

**Comment by Sarsmooch** [comment]
> I have the same issue, as i see the problem seems to occur somewhere in the caching, but i cannot say it in detail. The Agent::retrieveUsingCacheOrResolve() Method has the correct result, but the return value seems to be true.

I also looked into this, including the Agent::retrieveUsingCacheOrResolve()` method, I cleared the cache and everything I could, I hosted it on a server in order to check if it could be a problem on my machine, so far I haven't identified the real problem. Maybe later I look for more.

For now, what worked for me was simply modifying the echo to `{{ $session->agent->platform() ?: 'Unknown' }}`.

**Comment by crynobone** [comment]
@Sarsmooch can you submit a new bug report with Browser + Platform + User Agent string so we can debug this further.

**Comment by Rastian95** [comment]
I don't understand why remove functionalities instead of add new ones!?
 There is no more, isPhone(), isRobot(), robot(), deviceType(), device(), languages()
So now we need to extend the class and add these functions from jenssegers/agent? or what do you recommend to do?

**Comment by crynobone** [comment]
> So now we need to extend the class and add these functions from jenssegers/agent? or what do you recommend to do?

Feel free to add class/feature based on your requirements. `Agent` usage is limited to what's being used within Jetstream.

**Pull request merged**